scripting/weaponmodels.sp

```diff
@@ -284,11 +313,27 @@ public void OnClientPostAdminCheck(int client)
 
 public void OnClientSpawnPost(int client)
 {
-	// No spectators
-	if (GetClientTeam(client) < 2)
-	{
-		return;
-	}
+
+	//nmrih spectators have the same team id as non spectators: 0
+	//without this engine check the code will always end here in nmrih
+	//TODO: Find way to verify nmrih spectator status 
+	EngineVersion zeEngineVersion = GetEngineVersion();
+	if ( zeEngineVersion != view_as<EngineVersion>(ENGINE_NMRIH) ){
+		// No spectators
+		int clientteam = GetClientTeam(client);
+		if (clientteam < 2)
+		{
+			#if defined debug
```

Comment:
Remove the debug, In my oppinion, debug lines are very contextual, depends on where and when. It makes the code messy. I prefer not having any at all and should be added only when you are debuging a specific issue. It should be cleaned up after the issue is resolved.